internal/services/pubkey_service.go

```diff
@@ -20,57 +19,43 @@ func CreatePubkey(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// TODO: extract this into job queue
-	err := dao.WithTransaction(r.Context(), db.DB, func(tx dao.Transaction) error {
-		pkDao, err := dao.GetPubkeyDao(r.Context(), tx)
-		if err != nil {
-			return payloads.NewInitializeDAOError(r.Context(), "pubkey DAO", err)
-		}
+	pkDao, err := dao.GetPubkeyDao(r.Context())
+	if err != nil {
+		renderError(w, r, payloads.NewInitializeDAOError(r.Context(), "pubkey DAO", err))
+		return
+	}
+	pkrDao, err := dao.GetPubkeyResourceDao(r.Context())
+	if err != nil {
+		renderError(w, r, payloads.NewInitializeDAOError(r.Context(), "pubkey resource DAO", err))
+		return
+	}
 
-		err = pkDao.Create(r.Context(), payload.Pubkey)
-		if err != nil {
-			return payloads.NewDAOError(r.Context(), "create pubkey", err)
-		}
+	err = pkDao.Create(r.Context(), payload.Pubkey)
+	if err != nil {
+		renderError(w, r, payloads.NewDAOError(r.Context(), "create pubkey", err))
+		return
+	}
 
-		pkrDao, err := dao.GetPubkeyResourceDao(r.Context(), tx)
-		if err != nil {
-			return payloads.NewInitializeDAOError(r.Context(), "pubkey resource DAO", err)
-		}
-		pkr := models.PubkeyResource{
-			PubkeyID: payload.Pubkey.ID,
-			Provider: models.ProviderTypeAWS,
-		}
-		err = pkrDao.Create(r.Context(), &pkr)
-		if err != nil {
-			return payloads.NewDAOError(r.Context(), "create pubkey resource", err)
-		}
+	// create new resource with randomized tag
+	pkr := models.PubkeyResource{
+		PubkeyID: payload.Pubkey.ID,
+		Provider: models.ProviderTypeAWS,
+	}
+	pkr.RandomizeTag()
 
-		client := ec2.NewEC2Client(r.Context())
-		pkr.Handle, err = client.ImportPubkey(payload.Pubkey, pkr.FormattedTag())
-		if err != nil {
-			return payloads.NewAWSError(r.Context(), "import pubkey", err)
-		}
-		log.Trace().Msgf("Pubkey imported as '%s' with tag '%s'", pkr.Handle, pkr.FormattedTag())
-
-		err = pkrDao.Update(r.Context(), &pkr)
-		if err != nil {
-			var e *dao.MismatchAffectedError
-			if errors.As(err, &e) {
-				return payloads.NewNotFoundError(r.Context(), err)
-			} else {
-				return payloads.NewDAOError(r.Context(), "update pubkey resource", err)
-			}
-		}
+	// upload to cloud with a tag
+	client := ec2.NewEC2Client(r.Context())
+	pkr.Handle, err = client.ImportPubkey(payload.Pubkey, pkr.FormattedTag())
+	if err != nil {
+		renderError(w, r, payloads.NewAWSError(r.Context(), "import pubkey", err))
+		return
+	}
+	log.Trace().Msgf("Pubkey imported as '%s' with tag '%s'", pkr.Handle, pkr.FormattedTag())
 
-		return nil
-	})
+	// create resource with handle
+	err = pkrDao.Create(r.Context(), &pkr)
 	if err != nil {
-		var e *payloads.ResponseError
-		if errors.As(err, &e) {
-			renderError(w, r, e)
-		} else {
-			renderError(w, r, payloads.NewUnknownError(r.Context(), err))
-		}
+		renderError(w, r, payloads.NewDAOError(r.Context(), "create pubkey resource", err))
 		return
 	}
```

Comment:
This is now very much simplified (no transactions):

* Create pubkey
* Generate tag
* Upload to AWS
* Create pubkey resource (tag with handle)